milla/src/simulate.rs

```diff
@@ -242,15 +242,25 @@ pub(crate) fn check_interesting(
     Ok(())
 }
 
+pub(crate) fn is_significant(tile: &Tile, gas: usize) -> bool {
+    if tile.gases.values[gas] < REACTION_SIGNIFICANCE_MOLES {
+        return false;
+    }
+    if gas != GAS_AGENT_B && tile.gases.values[gas] / tile.gases.moles() < REACTION_SIGNIFICANCE_RATIO {
+        return false;
+    }
+    return true;
+}
```

Comment:
This is likely a minor (read: i doubt it's measurable) performance loss due to including the AGENT_B specific check making this unnecessarily complex for most uses (as LLVM is unlikely to inline this)

If gas is always constant, maybe make it a const generic argument so LLVM can make a distinct version of the function for agent b.
Untested but syntatically valid:
```rs

pub(crate) fn is_significant<const TARGET_GAS: usize>(tile: &Tile) -> bool {
    if tile.gases.values[TARGET_GAS] < REACTION_SIGNIFICANCE_MOLES {
        return false;
    }
    if gas != TARGET_GAS && tile.gases.values[TARGET_GAS] / tile.gases.moles() < REACTION_SIGNIFICANCE_RATIO {
        return false;
    }
    return true;
}
...

is_significant::<GAS_CARBON_DIOXIDE>(tile)


```

(also this elides some bounds checks for free which is cool)

Reply:
Understood. Not important enough to apply now; I'll take another look when I refactor reactions later on.